### ISSUE TITLE
Doesn't work with packaging version 22

### ISSUE THREAD

**Issue by lohedges** [conda-forge/pip-requirements-parser-feedstock#1]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

This package has no runtime pin against `packaging` and currently doesn't work against the latest version (22) since `packaging.version.LegacyVersion` has been removed.

```python
    from packaging.version import LegacyVersion
ImportError: cannot import name 'LegacyVersion' from 'packaging.version'
```

Please update the package to use the new API, or pin against a known range of working versions.

Many thanks.

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

**Comment by bollwyvl** [comment]
Packaging 22 broke _many_ things.

You'll likely want to report this on the upstream repo: we just repackage
here, and try to match upstream dependencies. Presumably pinning will get
you through until a compatible, or pinned, version is available.

**Comment by lohedges** [comment]
Thanks, it looks like it's already been reported there, although somewhat cryptically. I'll just pin for now and wait for any update to propagate its way through the system.

Cheers

**Comment by lohedges** [comment]
For reference, the issue thread is [here](https://github.com/nexB/pip-requirements-parser/issues/10). They have now pinned `packaging` in their `setup.cfg` [here](https://github.com/nexB/pip-requirements-parser/blob/5555b7a66abc9cb09746874378cc856cc67b3272/setup.cfg#L44-L45), but not in the `requirements.txt` [file](https://github.com/nexB/pip-requirements-parser/blob/main/requirements.txt). It sounds like this is a temporary fix until they work out how to support `packaging` version 22+.

**Comment by bollwyvl** [comment]
Welp, from a packaging perspective, `setup.cfg` is the law, and `pip check` will find the issue on the next release, which we can then update to match upstream.

**Comment by bollwyvl** [comment]
just shipped #2 (the bot did _not_ find `32.0.0`, so will keep track when the next release drops) which _does_ pin packaging. However, without a pin, conda/mamba might decide you'd rather have the newer `packaging`. Not sure if it makes sense to keep this open at this point, as the previous release isn't _broken_. It is possible to do repodata patches, but it's probably worth just keeping an eye on what happens.

**Comment by lohedges** [comment]
Thanks, I'm happy with whatever you decide. The problem is logged and someone else should be able to track it down necessary. For our purposes, this is just a tool that is used during the setup of our build process so it's very easy to pin and forget about it.
